packages/web/pages/user/my-account/questions.js

```diff
@@ -0,0 +1,309 @@
+	 */
+	const handlePagination = (page) => {
+		const { pathname, query } = router;
+		query.page = page;
+
+		router.push({
+			pathname,
+			query,
+		});
+	};
+
+	/**
+	 * Pushes new sort options to next/router
+	 *
+	 * @param {string} orderBy Grid column to sort items.
+	 * @param {('ASC'|'DESC')} order Sort order.
+	 * @returns {Promise<boolean>} Next router push
+	 */
+	const handleSortBy = (orderBy, order = currentSort.order || orderEnum.ASC) => {
+		const { pathname, query } = router;
+
+		delete query.page;
+
+		const shouldOrderAsc = order === orderEnum.DESC && currentSort.by !== orderBy;
+		query.order = shouldOrderAsc ? orderEnum.ASC : order;
+		query.orderBy = orderBy;
+
+		return router.push({
+			pathname,
+			query,
+		});
+	};
+
+	return (
+		<Container>
+			<Protected>
+				<UserProfile />
+				<MainContentContainer>
+					<Title align="left" noPadding noMargin>
+						{t('account:titles.questions')}
+					</Title>
+					<MainContent>
+						{questions.length ? (
+							<DataGrid
+								data={questions.map((question) => {
+									const { id, technology, user, status, created_at } = question;
+
+									return {
+										id,
+										Tecnologia: technology,
+										Usuário: user,
+										Status: (
+											<QuestionStatus status={status}>
+												{getQuestionStatusText(status)}
+											</QuestionStatus>
+										),
+										'Data da pergunta': dateToString(created_at),
+										Ações: (
+											<QuestionActions>
+												<IconButton
+													variant="gray"
+													aria-label="Question details"
+													onClick={() => openModal('questionDetails')}
+												>
+													<FiEye />
+												</IconButton>
+											</QuestionActions>
+										),
+									};
+								})}
+								hideItemsByKey={['id']}
+								currentPage={currentPage}
+								totalPages={totalPages}
+								totalItems={totalItems}
+								itemsPerPage={itemsPerPage}
+								currentOrder={currentSort.order}
+								sortOptions={sortOptions}
+								handlePagination={handlePagination}
+								handleSortBy={handleSortBy}
+								enablePagination
+							/>
+						) : (
+							<NoQuestions>{t('account:messages.noQuestionsToShow')}</NoQuestions>
+						)}
+					</MainContent>
+				</MainContentContainer>
+			</Protected>
+		</Container>
+	);
+};
+
+Questions.propTypes = {
+	questions: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
+	currentPage: PropTypes.number.isRequired,
+	totalPages: PropTypes.number.isRequired,
+	totalItems: PropTypes.number.isRequired,
+	currentSort: PropTypes.shape({
+		by: PropTypes.string,
+		order: PropTypes.string,
+	}),
+};
+
+Questions.defaultProps = {
+	currentSort: {},
+};
+
+Questions.getInitialProps = async (ctx) => {
+	const { query } = ctx;
+
+	const page = Number(query.page) || 1;
+
+	return {
+		questions: questionsMock,
+		currentPage: page,
+		totalPages: 1,
+		totalItems: 3,
+		currentSort: { by: query.orderBy, order: query.order },
```

Comment:
Was this sort feature supposed to be working at the moment? As far as I can see and doesn't order the values. It changes the URL query values, but the table is not updated. Not sure it this is something related to this feature or to the DataGrid...

cc @luizeboli

Reply:
Sadly ordering doesn't works for some routes.

Reply:
Hm...got it..so it's something related to the api...

Reply:
And there's no API for this issue yet.